### DISCUSSION TITLE
Assessment of the difficulty in porting CPU architecture for curl

### DISCUSSION THREAD

**Discussion by wangyuliu** [curl/curl#11841] [General]
Hello everyone! I am working on implementing a tool to assess the complexity of CPU architecture porting. It primarily focuses on RISC-V architecture porting. In fact, the tool may have an average estimate of various architecture porting efforts.My focus is on the overall workload and difficulty of transplantation in the past and future,even if a project has already been ported.As part of my dataset, I have collected the **curl** project. **I would like to gather community opinions to support my assessment. I appreciate your help and response!** Based on scanning tools, the porting complexity is determined to be simple, with a small amount of code related to the CPU architecture in the project.  Is this assessment accurate?Do you have any opinions on personnel allocation and consumption time？ I look forward to your help and response.

**Comment by bagder** [comment]
Building curl is easy. The particular CPU choice is hardly ever a problem as long as you have a properly working C89 compiler for it. The challenge that might arise is if your operating system is not already supported.

For something like Linux on a new CPU architecture, just invoke configure and build it with gcc or clang. No problemos.

**Comment by rsbeckerca** [comment]
If I can chime in from experience, getting the `#includes`, `#defines`, and compiler options can be challenging. But it is usually doable.
